internal/replacements/replacements.go

```diff
@@ -153,7 +163,7 @@ func processGroupNeedsRemoval(cluster *fdbv1beta2.FoundationDBCluster, pod *core
 
 	_, desiredProcessGroupID := cluster.GetProcessGroupID(processGroupStatus.ProcessClass, idNum)
 	if processGroupStatus.ProcessGroupID != desiredProcessGroupID {
-		logger.Info("Replace process group",
+		logger.Info("Process group requires removal",
```

Comment:
What is the benefit of changing those logs? That could break some existing dashboard/alerts based on this logging statement. In general I would favour the term `Process group requires replacement`.

Reply:
The function is also used by updatePods. `"Replace process group"` is a bit misleading log message from the updatePods reconciler, which will not perform an update (deletion) in that case.
We could log it higher up in the ReplaceMisconfiguredProcessGroups function (if needsRemoval == true).

Regarding replacement vs removal. Would it make sense to refactor the methods from processGroupNeeds**Removal*** to processGroupNeeds**Replacement*** and replace (😄) other occurrences of removal with replacement?

Reply:
> The function is also used by updatePods. "Replace process group" is a bit misleading log message from the updatePods reconciler, which will not perform an update (deletion) in that case.
We could log it higher up in the ReplaceMisconfiguredProcessGroups function (if needsRemoval == true).

It's probably fine enough to use the combination of this log message and the `ReplaceMisconfiguredProcessGroups` reconciler in the log message to figure out that those messages mean that the process group gets replaced.

> Regarding replacement vs removal. Would it make sense to refactor the methods from processGroupNeedsRemoval* to processGroupNeedsReplacement* and replace (😄) other occurrences of removal with replacement?

I think that would make sense but we should do that in a different PR.